### ISSUE TITLE
Protocol design goals for resource usage

### ISSUE THREAD

**Issue by Christewart** [discreetlogcontracts/dlcspecs#64]
I thought I would enumerate somethings things I think we should keep in mind when writing our first versions of the protocol. The huge caveat here is correctness and security of the protocol come first, here are some secondary design goals we should keep in mind when writing these specs. The overall design goal I want to convey here is that we should attempt to be a  ["thin" protocol rather than a "fat protocol"](https://www.usv.com/writing/2016/08/fat-protocols/). Thin protocols give more flexibility to application developers and also minimizes trust assumptions in 3rd parties. 

1. Minimize interactivity in the protocol (security & resource usage)
2. Minimize bandwidth needed (resource usage, somewhat related to 1)
3. ~When possible, shift resource responsibilities to oracles rather than clients -- clients must retain a simple way to validate oracles (think relationship between miners and full nodes)~

Again, I don't have any concrete proposals, but we should have these goals in mind when we have choices to make in the protocol. Is there anything that should be added?

**Comment by Christewart** [comment]
4. Make conservative assumptions about the blockchain (this is needed for security reasons, but also for usefulness/portability across chains)

**Comment by benthecarman** [comment]
I definitely agree with the flexibility, we should give as much functionality to the users and let them decide what is best for them and not force certain assumptions on into the protocol.

**Comment by Tibo-lg** [comment]
I agree with 1, 2 and 4. For 3 I'm not sure what you mean and how that translate in the protocol specifications. The oracle plays a quite specific role and I don't think it can take more responsibility without changing its nature.

> I definitely agree with the flexibility, we should give as much functionality to the users and let them decide what is best for them and not force certain assumptions on into the protocol.

@benthecarman I don't especially disagree, but I feel there is a compromise between flexibility and efficiency. I might be wrong, but in my view the more flexible you are, the more complex the protocol, and vice versa.

**Comment by Christewart** [comment]
> I agree with 1, 2 and 4. For 3 I'm not sure what you mean and how that translate in the protocol specifications. The oracle plays a quite specific role and I don't think it can take more responsibility without changing its nature.

Thinking about this more I think you are right. It doesn't make as much sense in this context.
